bundle/compliance/cis_gcp/rules/cis_6_4/rule.rego

```diff
@@ -0,0 +1,38 @@
+package compliance.cis_gcp.rules.cis_6_4
+
+import data.compliance.lib.common
+import data.compliance.policy.gcp.data_adapter
+import future.keywords.if
+import future.keywords.in
+
+finding = result if {
+	data_adapter.is_sql_instance
+	data_adapter.resource.databaseVersion in sql_db_versions
+
+	result := common.generate_result_without_expected(
+		common.calculate_result(ssl_is_required),
+		data_adapter.resource,
+	)
+}
+
+ssl_is_required if {
+	data_adapter.resource.data.settings.ipConfiguration.requireSsl == true
+} else = false
+
+# rule only applies to: 
+# Postgresql, MySql generation 1, MySql generation 2 and SQL Server 2017 instances.
+sql_db_versions := {
+	"SQLSERVER_2017_STANDARD",
+	# Gen 1
+	"MYSQL_5_6",
+	"MYSQL_5_7",
+	# Gen 2
+	"MYSQL_8_0",
+	"POSTGRES_9_6",
+	"POSTGRES_10",
+	"POSTGRES_11",
+	"POSTGRES_12",
+	"POSTGRES_13",
+	"POSTGRES_14",
+	"POSTGRES_15",
+}
```

Comment:
> This recommendation is applicable for Postgresql, MySql generation 1, MySql generation 2 and SQL Server 2017 instances.

- all of postgresql 
- mysql gen 1 and 2 
- sqlserver 2017

Reply:
another approach:
```rego
is_relevant_sql_instance if {
	startswith(data_adapter.resource.data.databaseVersion, "POSTGRES")
} else = false

is_relevant_sql_instance if {
	data_adapter.resource.data.databaseVersion in {"MYSQL_5_6", "MYSQL_5_7", "MYSQL_8_0"}
} else = false

is_relevant_sql_instance if {
	startswith(data_adapter.resource.data.databaseVersion, "SQLSERVER_2017")
} else = false
```

wdyt?

Reply:
cool, I prefer the current approach.
Will the size of a blacklist be the same?
I'm concerned about a scenario in which a new version is supported but we'll not evaluate those instances (e.g new postgres version).

Reply:
the current approach uses the current values in GCP but doesn't account for newly added values as we can't tell what those would be in advance. (like in `postgres`). the second approach does account for it though.

Reply:
so let's go with the second approach then